pkg/controllers/leaderworkerset_controller.go

```diff
@@ -96,13 +100,31 @@ func (r *LeaderWorkerSetReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	log := ctrl.LoggerFrom(ctx).WithValues("leaderworkerset", klog.KObj(lws))
 	ctx = ctrl.LoggerInto(ctx, log)
 
-	partition, replicas, err := r.rollingUpdateParameters(ctx, lws)
+	leaderSts, err := r.getLeaderStatefulSet(ctx, lws)
+	if err != nil {
+		log.Error(err, "Fetching leader statefulset")
+		return ctrl.Result{}, err
+	}
+
+	if err := r.createControllerRevisionIfNonExist(ctx, leaderSts, lws); err != nil {
+		log.Error(err, "Creating controller revision")
+		return ctrl.Result{}, err
+	}
+
+	lwsUpdated, err := r.leaderWorkerSetUpdated(ctx, leaderSts, lws)
+	if err != nil {
+		log.Error(err, "Validating if LWS has been updated")
+		return ctrl.Result{}, err
+	}
+
+	templateHash := getLeaderWorkerTemplateHash(leaderSts, lws, lwsUpdated)
```

Comment:
we should not create a template hash here, it should come from the revision returned as suggested in line 109 above

Reply:
Don't we still need a hashing function that treats `NetworkConfig == nil` and `NetworkConfig.SubdomainShared` as the same though? 

When updating from a version that doesn't have networkConfig the steps would be:
Create a controller revision where the NetworkConfig si nil. When checking if LWS was updated, it will determine that it isn't because the hash is the same. 

Another change is made, similar to https://github.com/kubernetes-sigs/lws/issues/240#issuecomment-2452679190 triggering the lws webhook, and defaulting NetworkConfig. A new controller revision will be created, as the data will now be different. Now, the hash won't match the hash that is in the leaderSts, triggering a rolling update when it isn't supposed to.

Reply:
We are not comparing hashes anymore though, the hash is just a key to the controller revision. We depend on semantic diff of the revisions.